server/src/main/java/com/distkv/server/AbstractService.java

```diff
@@ -0,0 +1,82 @@
+package com.distkv.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractService implements Service {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractService.class);
+
+  private static final boolean[][] stateMap =
```

Comment:
Could you give more detail of this stateMap?

Reply:
the lifycycle of service is composed of `UNINITED`, `INITED`, `RUNNING`, 'STOPPED'. the `UNINITED` stauts can evolve to `INITED`, but can't `RUNNING`. Similarly, the `RUNNING` can evolve to `STOPPED` but `UNINITED` and `INITED`.

Reply:
OK, I see. But it seems a bit tricky.